tasks/go.py

```diff
@@ -159,7 +163,14 @@ def ineffassign(ctx, targets):
         # as comma separated tokens in a string
         targets = targets.split(',')
 
-    ctx.run("ineffassign " + " ".join(targets))
+    files = []
+    for target in targets:
```

Comment:
Technically targets could be loose files too. So before you start walking you might want to make sure it's a directory, and if it is a file, then just compare it directly with the whitelist.